src/io/aleph/dirigiste/Pool.java

```diff
@@ -209,8 +209,7 @@ public boolean take(AcquireCallback<V> c, boolean skipToFront) throws RejectedEx
     private final Set<V> _destroyedObjects = Collections.synchronizedSet(Collections.newSetFromMap(new WeakHashMap<>()));
     private final ConcurrentHashMap<V,Long> _start = new ConcurrentHashMap<V,Long>();
     final ConcurrentHashMap<K,Queue> _queues = new ConcurrentHashMap<>();
-    private final ConcurrentHashMap.KeySetView<K, Boolean> _lockedQueues = ConcurrentHashMap.newKeySet();
-
+    private final ConcurrentHashMap<K, Integer> _lockedQueues = new ConcurrentHashMap<>();
```

Comment:
Should probably be renamed to something like `_queueLockCount` now, if it's not a set of queues.